### PR TITLE
feat(generatedAnswerAnalyticsAction): stream id

### PR DESCRIPTION
**Pull request by dmgauthier**
SVCC-4021

# Managing the answer api flow rga components custom events

## With the answer api flow
The answer id will be sent to the custom event
image

## With the search Api flow
The search api stream ID will be sent with the custom event
image

## Description

It was decided that we should send a distinct key in the custom event when using the answer api stream id. So I modified the selector to return the `answerAPIEnabled` is boolean that will help determine what key to use when sending the event. Not sure if this could have been done in a better way.

### CODE REVIEW

File: packages/headless/src/controllers/knowledge/generated-answer/headless-answerapi-generated-answer.ts
Comment (dmgauthier) [inline packages/headless/src/controllers/knowledge/generated-answer/headless-answerapi-generated-answer.ts:1]: This modification is out of scope of the goal described in the PR.

The conditions changes here will make it so we are not calling the stream uselessly.
A set of tests should be implemented before going GA with the feature.

File: patches/coveo.analytics+2.30.38.patch [hunk 1]
```diff
@@ -1,17 +1,8 @@
 diff --git a/node_modules/coveo.analytics/dist/browser.mjs b/node_modules/coveo.analytics/dist/browser.mjs
-index 3de0ed7..b7d2c0e 100644
+index 78efccc..7702f2d 100644
 --- a/node_modules/coveo.analytics/dist/browser.mjs
 +++ b/node_modules/coveo.analytics/dist/browser.mjs
-@@ -1093,8 +1093,6 @@ class NoopAnalyticsClient {
-     }
- }
- 
--const fetch$1 = window.fetch;
```
Comment (louis-bompart) [inline patches/coveo.analytics+2.30.38.patch:9]: We need this: Otherwise it will break on NodeJS

**Review by louis-bompart — CHANGES_REQUESTED** [review]

File: patches/coveo.analytics+2.30.38.patch (hunk 1 above)
Comment (dmgauthier) [reply patches/coveo.analytics+2.30.38.patch:9]: It indeed does whenever the code is executed outside the browser. :+1:

File: packages/headless/src/features/generated-answer/generated-answer-selectors.test.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    afterAll(() => {
+      jest.clearAllMocks();
+    });
+    it('returns the answerId if an answer configuration id is in state', () => {
+      const state = {
+        ...(streamAnswerAPIStateMock as any),
+        generatedAnswer: {
+          answerConfigurationId: 'answerConfigurationId',
+        },
+      } as any;
```
Comment (louis-bompart) [inline packages/headless/src/features/generated-answer/generated-answer-selectors.test.ts:27]: Does `unknown` work instead?

**Review by louis-bompart — APPROVED** [review]
lgtm, tho please try to use unknown instead of any

File: packages/headless/src/features/generated-answer/generated-answer-selectors.test.ts (hunk 2 above)
Comment (dmgauthier) [reply packages/headless/src/features/generated-answer/generated-answer-selectors.test.ts:27]: no but `Partial<SearchAppState>` does
